control/autoware_lane_departure_checker/src/lane_departure_checker_node/utils.cpp

```diff
@@ -163,6 +165,64 @@ std::vector<LinearRing2d> createVehicleFootprints(
   return vehicle_footprints;
 }
 
+lanelet::ConstLanelets getCandidateLanelets(
+  const lanelet::ConstLanelets & route_lanelets,
+  const std::vector<LinearRing2d> & vehicle_footprints)
+{
+  lanelet::ConstLanelets candidate_lanelets;
+
+  // Find lanes within the convex hull of footprints
+  const auto footprint_hull = createHullFromFootprints(vehicle_footprints);
+
+  for (const auto & route_lanelet : route_lanelets) {
+    const auto poly = route_lanelet.polygon2d().basicPolygon();
+    if (!boost::geometry::disjoint(poly, footprint_hull)) {
+      candidate_lanelets.push_back(route_lanelet);
+    }
+  }
```

Comment:
seems like a good place to use std::copy_if

Reply:
performance doesn't change and prefer original one so ignore it :smile_cat: